### PR TITLE
Add guidance on using U2F key handles with appid extension

### LINKED ISSUE #1229: Document that U2F key handle = credential ID
The spec currently offers little guidance on how to use U2F credentials in WebAuthn via the `appid` extension; in particular, we don't mention that U2F key handles are usable as WebAuthn credential IDs without modification.

### PR DESCRIPTION
**Pull request by emlun**
Fixes #1229

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/1230.html" title="Last updated on Jun 19, 2019, 5:17 PM UTC (414e265)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/1230/db94f4a...414e265.html" title="Last updated on Jun 19, 2019, 5:17 PM UTC (414e265)">Diff</a>

### CODE REVIEW

**Comment by emlun** [comment]
I'm not entirely happy with how the added text talks about "U2F credentials" while the preceding paragraph is about "credentials created using the legacy FIDO JavaScript APIs", but I think this could be the way to go after all. I experimented with abstracting it as "legacy credentials" or "AppID credentials", but since U2F is currently our only backwards compatibility I felt like that makes it unnecessarily vague and half defeats the purpose of adding specific guidance on how to authenticate U2F credentials.

File: index.bs [hunk 1]
```diff
@@ -4558,6 +4558,23 @@ FIDO APIs use an alternative identifier for [=[RPS]=] called an |AppID|
 that identifier. Without this extension, they would need to be re-registered in
 order to be [=scoped=] to an [=RP ID=].
 
+In addition to setting the {{AuthenticationExtensionsClientInputs/appid}} extension input,
+using this extension requires some additional processing by the [=[RP]=]
+in order to [=authentication|authenticate=] U2F credentials:
```
Comment (equalsJeffH) [inline index.bs]: suggest:
in order to allow users to [=authentication|authenticate=] using their registered U2F credentials:

File: index.bs [hunk 2]
```diff
@@ -4558,6 +4558,23 @@ FIDO APIs use an alternative identifier for [=[RPS]=] called an |AppID|
...
+
+ 1. List the desired U2F credentials in the {{PublicKeyCredentialRequestOptions/allowCredentials}} option
+     of the {{CredentialsContainer/get()}} method:
+
+     - Set the {{PublicKeyCredentialDescriptor/type}} members to {{PublicKeyCredentialType/public-key}}.
+     - Set the {{PublicKeyCredentialDescriptor/id}} members to the respective U2F key handles of the desired credentials.
+
+     {{PublicKeyCredentialRequestOptions/allowCredentials}} MAY contain a mixture
+     of both WebAuthn [=credential IDs=] and U2F key handles;
+     setting the {{AuthenticationExtensionsClientInputs/appid}} does not prevent the user from using a WebAuthn credential.
```
Comment (equalsJeffH) [inline index.bs]: suggest:
stating the {{AuthenticationExtensionsClientInputs/appid}} via this extension does not prevent the user from using a WebAuthn-registered credential that is scoped to the [=RP ID=] stated in  {{PublicKeyCredentialRequestOptions/rpId}}.

File: index.bs [hunk 3]
```diff
@@ -4558,6 +4558,23 @@ FIDO APIs use an alternative identifier for [=[RPS]=] called an |AppID|
...
+
+ 1. When [verifying the assertion](#sctn-verifying-assertion), expect that the <code>[=rpIdHash=]</code>
+     MAY be the hash of the |AppID| instead of the [=RP ID=].
```
Comment (equalsJeffH) [inline index.bs]: I wonder whether we ought to add a Note: referencing this extension to #sctn-verifying-assertion on the appropriate step(s)?

Comment (emlun) [reply index.bs]: There's a link to `#sctn-verifying-assertion` right there. :) But it's a good idea to link directly to the relevant step, so I added an anchor to step 6 and updated the links to point directly there. Seem good?

Comment (equalsJeffH) [reply index.bs]: sorry, I wasn't clear: what I meant was to add a Note to that 'rp-op-verifying-assertion-step-rpid-hash' step in #sctn-verifying-assertion that points back over here and says something like "if you use the AppId extension, see here for additionall processing details..."

Comment (equalsJeffH) [reply index.bs]: > so I added an anchor to step 6 and updated the links to point directly there

nice :)

Comment (emlun) [reply index.bs]: I added a note to the RP assertion verification steps, does that look good?

**Comment by equalsJeffH** [comment]
I will review this today.

**Pull request merged**